orte/orted/pmix/pmix_server_register_fns.c

```diff
@@ -324,6 +288,43 @@ int orte_pmix_server_register_nspace(orte_job_t *jdata)
         kv->data.integer = pptr->name.vpid;
         opal_list_append(pmap, &kv->super);
 
+        /* local size - same for all on host, but not same across namespace */
+        kv = OBJ_NEW(opal_value_t);
+        kv->key = strdup(OPAL_PMIX_LOCAL_SIZE);
+        kv->type = OPAL_UINT32;
+        kv->data.uint32 = jdata->num_local_procs;
+        opal_list_append(pmap, &kv->super);
+
+        /* pass our node ID - same for all on host, but not same across namespace */
+        kv = OBJ_NEW(opal_value_t);
+        kv->key = strdup(OPAL_PMIX_NODEID);
+        kv->type = OPAL_UINT32;
+        kv->data.uint32 = node->index;
+        opal_list_append(pmap, &kv->super);
+
+        /* construct the list of peers for transmission */
+        /* - same for all on host, but not same across namespace */
+        if (NULL != list) {
+            tmp = opal_argv_join(list, ',');
```

Comment:
I don't think this is correct - isn't this list the one we generated above that is specific to this one node? I think you'll need to regenerate this for each proc using the peers on just its node.